types/qunit/index.d.ts

```diff
@@ -560,11 +560,14 @@ declare global {
         }
 
         type TestFunctionCallback = (assert: Assert) => void | Promise<void>;
-        type EachFunction = <T>(
-            name: string,
-            dataset: T[] | { [key: string]: T },
-            callback: (assert: Assert, data: T) => void,
-        ) => void;
+
+        interface EachFunction {
```

Comment:
Sorry if this is obvious, but can you explain the choice between `interface` vs `type` here? Is this needed for the below? Or does it ease ember-qunit? Or something else?

Reply:
@Krinkle it's so that in ember-qunit we can just add a new callable signature by extending the interface

```typescript
namespace QUnit {
  interface EachFunction {
     <TC extends TestContext, T>(
        name: string,
        dataset: T[]
        callback: (this: TC, assert: Assert, data: T) => void | Promise<unknown>
      ): void;
  }
}

```